### PR TITLE
CIV-7098 Java upgraded to version 17

### PR DESCRIPTION
**Pull request by maciejmodzelewski**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-7098

### Change description ###
Java upgraded to version 17, along with some dependencies. Notable ones:
- AppInsights: needed to be updated
- Gradle: needed to be updated
- Checkstyle: needed to be updated and some related files to be modified to fit the rules and some of its changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -258,13 +261,6 @@ task runDJTests(type: Exec, description: 'Runs DJ Api functional tests.') {
   commandLine '/usr/bin/yarn', '--silent', 'test:api-dj'
 }
 
-task runCosTests(type: Exec, description: 'Runs DJ Api functional tests.') {
-  onlyIf {
-    return System.env.ENVIRONMENT == 'preview'
```
Comment (vasudevganesanhmcts) [inline build.gradle:263]: this is removed by accident please revert looking at master

File: build.gradle [hunk 2]
```diff
@@ -294,7 +290,7 @@ task smoke(description: 'Runs the smoke tests.') {
 
 task functional(description: 'Runs the functional tests.') {
   dependsOn(inStrictOrder(awaitApplicationReadiness, pullTestAsset, installDependencies, checkDependenciesIntegrity,
-    runCosTests, runDJTests, runSpecApiTests, runSdoApiTests, runApiTests))
```
Comment (vasudevganesanhmcts) [inline build.gradle:297]: this is removed by accident please revert looking at master

Comment (maciejmodzelewski) [reply build.gradle]: The missing code restored

**Review by miguelMolina3691 — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Comment by filippo-possenti-version1** [comment]
Last commit was just to fix a checkstyle. This happens because the change in version solved some issues so there's differences in the way these conventions are enforced. No need to re-approve for this.

**Pull request merged**